test/IdRegistry/IdRegistry.t.sol

```diff
@@ -1430,6 +1430,375 @@ contract IdRegistryTest is IdRegistryTestSuite {
         assertEq(idRegistry.recoveryOf(2), address(0));
     }
 
+    /*//////////////////////////////////////////////////////////////
+                        RECOVER FOR TESTS
+    //////////////////////////////////////////////////////////////*/
+
+    function testFuzzRecoverFor(
+        address caller,
+        uint256 recoveryPk,
+        uint256 toPk,
+        address from,
+        uint40 _recoveryDeadline,
+        uint40 _toDeadline
+    ) public {
+        recoveryPk = _boundPk(recoveryPk);
+        toPk = _boundPk(toPk);
+        address recovery = vm.addr(recoveryPk);
+        address to = vm.addr(toPk);
+        vm.assume(from != to);
+        vm.assume(recovery != to);
+
+        uint256 recoveryDeadline = _boundDeadline(_recoveryDeadline);
+        uint256 toDeadline = _boundDeadline(_toDeadline);
+        uint256 fid = _registerWithRecovery(from, recovery);
+
+        bytes memory recoverySig = _signTransfer(recoveryPk, fid, to, recoveryDeadline);
+        bytes memory toSig = _signTransfer(toPk, fid, to, toDeadline);
+
+        assertEq(idRegistry.idCounter(), 1);
+        assertEq(idRegistry.idOf(from), 1);
+        assertEq(idRegistry.idOf(to), 0);
+
+        address recoveryBefore = idRegistry.recoveryOf(fid);
+
+        vm.expectEmit();
+        emit Recover(from, to, 1);
+        vm.expectEmit();
+        emit Transfer(from, to, 1);
+        vm.prank(caller);
+        idRegistry.recoverFor(from, to, recoveryDeadline, recoverySig, toDeadline, toSig);
+
+        assertEq(idRegistry.idCounter(), 1);
+        assertEq(idRegistry.idOf(from), 0);
+        assertEq(idRegistry.idOf(to), 1);
+        assertEq(idRegistry.recoveryOf(fid), recoveryBefore);
+    }
+
+    function testFuzzRecoverForRevertsInvalidRecoverySig(
+        address caller,
+        uint256 recoveryPk,
+        uint256 toPk,
+        address from,
+        uint40 _recoveryDeadline,
+        uint40 _toDeadline
+    ) public {
+        recoveryPk = _boundPk(recoveryPk);
+        toPk = _boundPk(toPk);
+        address recovery = vm.addr(recoveryPk);
+        address to = vm.addr(toPk);
+        vm.assume(from != to);
+        vm.assume(recovery != to);
+
+        uint256 recoveryDeadline = _boundDeadline(_recoveryDeadline);
+        uint256 toDeadline = _boundDeadline(_toDeadline);
+        uint256 fid = _registerWithRecovery(from, recovery);
+
+        bytes memory recoverySig = _signTransfer(recoveryPk, fid, to, recoveryDeadline + 1);
+        bytes memory toSig = _signTransfer(toPk, fid, to, toDeadline);
+
+        assertEq(idRegistry.idCounter(), 1);
+        assertEq(idRegistry.idOf(from), 1);
+        assertEq(idRegistry.idOf(to), 0);
+
+        vm.prank(caller);
+        vm.expectRevert(Signatures.InvalidSignature.selector);
+        idRegistry.recoverFor(from, to, recoveryDeadline, recoverySig, toDeadline, toSig);
+
+        assertEq(idRegistry.idCounter(), 1);
+        assertEq(idRegistry.idOf(from), 1);
+        assertEq(idRegistry.idOf(to), 0);
```

Comment:
nit: it might be worth asserting that the recovery address is untouched for all failed cases